python/semantic_kernel/agents/channels/agent_channel.py

```diff
@@ -43,7 +43,24 @@ def invoke(
             agent: The agent to interact with.
 
         Returns:
-            A async iterable of a bool, ChatMessageContent.
+            An async iterable of a bool, ChatMessageContent.
+        """
+        ...
+
+    @abstractmethod
+    def invoke_stream(
+        self,
+        agent: "Agent",
+        history: "list[ChatMessageContent]",
```

Comment:
Question: why do we not want to use the `ChatHistory` type? (I see `list[ChatMessageContent]` used elsewhere too instead of `ChatHistory`)

Reply:
it seems to be an agent framework decision stemming from dotnet. I can do an overhaul at some point to use chat history, where applicable. There is some work in the future about support kernel args for agents that may get us there (it's experimental so things are in flux).

Reply:
To add some more context: we have different agent channels, one channel for a ChatCompletionAgent (SK flavor of an agent that _can_ use a ChatHistory), and then we have OpenAI assistant agents, which introducing a chat history to that type of channel may not make sense -- OpenAI Assistants don't have a `ChatHistory` type abstraction, so we manage list[ChatMessageContent], and use it to create messages on the Assistant thread.